CMakeLists.txt

```diff
@@ -38,132 +41,23 @@ set(NOVELRT_RESOURCES
 
 file(COPY ${NOVELRT_RESOURCES} DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
 
-if(${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm") #TODO: check if the OPENGL_VERSION flag is actually still needed with the new setup
-    file(GLOB CORECLR lib/CoreCLR/LinuxARM32/*)
-    add_compile_definitions(OPENGL_VERSION=2)
-else()
-    add_compile_definitions(OPENGL_VERSION=3)
-endif()
-
-file(COPY ${CORECLR} DESTINATION lib/CoreCLR/Active)
-
-find_package(Freetype REQUIRED)
-find_package(glad CONFIG REQUIRED)
-find_package(glm CONFIG REQUIRED)
-find_package(SDL2 CONFIG REQUIRED)
-find_package(sdl2-image CONFIG REQUIRED)
-find_package(sdl2-mixer CONFIG REQUIRED)
-find_package(spdlog CONFIG REQUIRED)
-
-# lua doesn't provide native cmake support
-find_path(LUA_INCLUDE_DIR lua.h)
-include_directories(${LUA_INCLUDE_DIR})
-
-find_library(LUA_LIBRARY lua)
-link_libraries(${LUA_LIBRARY})
+include_directories(src)
 
 find_package(Doxygen QUIET)
 
-set(NOVELRT_HEADERS
-  src/NovelWorldObject.h
-  src/GeoBounds.h
-  src/GeoVector.h
-  src/GraphicsCharacterRenderData.h
-  src/KeyCode.h
-  src/Lazy.h
-  src/NovelAudioService.h
-  src/NovelBasicFillRect.h
-  src/NovelBasicInteractionRect.h
-  src/NovelCommonArgs.h
-  src/NovelDebugService.h
-  src/NovelImageRect.h
-  src/NovelInteractionObject.h
-  src/NovelInteractionService.h
-  src/NovelLayeringService.h
-  src/NovelLoggingService.h
-  src/NovelObjectSortComparison.h
-  src/NovelRenderingService.h
-  src/NovelRenderObject.h
-  src/NovelUtilities.h
-  src/NovelRunner.h
-  src/NovelStepTimer.h
-  src/NovelTextRect.h
-  src/RGBAConfig.h
-  src/wrapper.hpp
-  src/NovelCamera.h
-  src/GeoMatrix4.h
-  src/CameraViewChangedEventArgs.h
-  src/ShaderProgram.h
-  src/CameraBlock.h
-  src/NovelRT.h)
-
-set(NOVELRT_SOURCES
-        src/NovelWorldObject.cpp
-  src/GeoBounds.cpp
-  src/NovelAudioService.cpp
-  src/NovelBasicFillRect.cpp
-  src/NovelBasicInteractionRect.cpp
-  src/NovelDebugService.cpp
-  src/NovelImageRect.cpp
-  src/NovelInteractionObject.cpp
-  src/NovelInteractionService.cpp
-  src/NovelLayeringService.cpp
-  src/NovelLoggingService.cpp
-  src/NovelRenderingService.cpp
-  src/NovelRenderObject.cpp
-  src/NovelRunner.cpp
-  src/NovelStepTimer.cpp
-  src/NovelTextRect.cpp
-  src/RGBAConfig.cpp
-  src/NovelCamera.cpp)
-
-set(NOVELRT_LINK_LIBRARIES
-  Freetype::Freetype
-  glad::glad
-  glm
-  SDL2::SDL2-static
-  SDL2::SDL2_image
-  SDL2::SDL2_mixer
-  SDL2::SDL2main
-  spdlog::spdlog
-)
-
-if(NOT WIN32)
-    set(NOVELRT_LINK_LIBRARIES
-      ${NOVELRT_LINK_LIBRARIES}
-      m
-    )
-endif()
-
-set(NOVELRT_EXECUTABLE_HEADERS
-  ${NOVELRT_HEADERS}
-)
-
 set(NOVELRT_EXECUTABLE_SOURCES
```

Comment:
Shouldn't this (and others) be moved to the respective projects rather than be in the root?

Reply:
I think any remnants are Doxygen specific/have just not been cleaned out yet or I mistook them for shared settings such as `set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)`.